### ISSUE TITLE
"ghcjs-base-0.1.0.0 depends on aeson-0.8.0.2 which failed to install."

### ISSUE THREAD

**Issue by sboosali** [ghcjs/ghcjs#303]
the build failure, after two hours:

    ghcjs-0.1.0-7.8.3.bin: Data/Aeson/TH.hs:2:46-60: You can't use Template Haskell with a profiled compiler
    Usage: For basic information, try the `--help' option.
    cabal: Error: some packages failed to install:
    aeson-0.8.0.2 failed during the building phase. The exception was:
    ExitFailure 1
    ghcjs-base-0.1.0.0 depends on aeson-0.8.0.2 which failed to install.

my cabal configuration is not the default:

    $ cabal user-config diff
    + documentation: True
    + hyperlink-source: True
    + library-profiling: True
    + profiling: True
    - user-install: True
    + user-install: False

is that related to my `library-profiling: True`? is the `Data/Aeson/TH.hs` module really ghcjs's  only dependency with a TemplateHaskell extension? (https://hackage.haskell.org/package/aeson-0.8.0.2/docs/src/Data-Aeson-TH.html). Or is the issue something else?

I successfully ran `cabal install aeson-0.8.0.2`, with library profiling enabled.

The only other time I've seen this error is with template Haskell in a doctest, because of its use of GHCi.

I followed the instructions:

    $ git clone https://github.com/ghcjs/ghcjs-prim.git
    $ git clone https://github.com/ghcjs/ghcjs.git
    $ time cabal install --user  ./ghcjs ./ghcjs-prim  --reorder-goals --max-backjumps=-1

after first having had run:

    $ brew install automake
    $ brew install nodes

and I had to use sudo, if relevant:

    $ sudo time ghcjs-boot --dev

the executables are on my path and have the right version:

    program         version    location                             
                                                                   
    ghcjs           0.1.0      /Users/sambo/.cabal/bin/ghcjs        
    ghcjs-pkg       0.1.0      /Users/sambo/.cabal/bin/ghcjs-pkg    
    ghcjs-run       -          /Users/sambo/.cabal/bin/ghcjs-run    
    ghc             7.8.3      /usr/local/bin/ghc                   
    ghc-pkg         -          /usr/local/bin/ghc-pkg               
    cabal           1.22.0.0   /Users/sambo/.cabal/bin/cabal        
    node            v0.10.36   /usr/local/bin/node                  
    haddock-ghcjs   -          /Users/sambo/.cabal/bin/haddock-ghcjs
    git             -          /usr/local/bin/git                   
    alex            -          /Users/sambo/.cabal/bin/alex         
    happy           -          /Users/sambo/.cabal/bin/happy        
    tar             -          /usr/bin/tar                         
    cpp             -          /usr/bin/cpp                         
    bash            -          /bin/bash                            
    autoreconf      -          /usr/local/bin/autoreconf            
    make            -          /usr/bin/make                          

I'm on OS X 10.9, if relevant:

    $ sw_vers
    ProductName:	Mac OS X
    ProductVersion:	10.9.5
    BuildVersion:	13F34

thanks! Can't wait to try out some Haskell in the browser :-)

**Comment by luite** [comment]
The problem is the 'profiling: True' line. Unfortunately the GHC API does not allow Template Haskell to run when the compiler is built with profiling, even though for GHCJS it would actually not lead to problems.

I thought there was a check in `ghcjs-boot` for this, so that it at least wouldn't waste as much time. Profiling does make the program a lot slower, so when you reinstall without executable profiling, it should boot much faster.

**Comment by sboosali** [comment]
yup, I thought it would be okay because I read that you just added support for call centers. But of course, that's in the libraries that are built with ghcjs, not building ghcjs itself.

**Comment by sboosali** [comment]
also, is it normal for ghcjs-boot to need `sudo`?

**Comment by luite** [comment]
no it shouldn't, and it will likely make things worse later if you do use sudo. the default location for the libraries is under `~/.ghcjs/`, make sure that's writable.

You can change the location by changing the ghcjs wrapper scripts (the `ghcjs` program is a script that passes the library location to the `ghcjs` program. `ghcjs-boot` asks ghcjs for the location). After booting, ghcjs does not need to write to the global installation dir anymore.
